src/processors/universal.ts

```diff
@@ -370,6 +369,136 @@ class UniversalProcessor extends BaseOrderProcessor {
     return unlockTx;
   }
 
+  private unlockOrder(
+    orderId: string,
+    order: OrderData,
+    context: OrderProcessorContext
+  ) {
+    const giveChain = order.give.chainId;
+    // filling batch queue
+    let orderIds = this.unlockBatchesOrderIdMap.get(giveChain);
+    if (!orderIds) {
+      orderIds = new Set();
+    }
+    orderIds.add(orderId);
+    this.unlockBatchesOrderIdMap.set(giveChain, orderIds);
+    this.ordersDataMap.set(orderId, order);
+
+    if (this.unlockBatchesLockedStatus.get(giveChain) === true) {
```

Comment:
Already discussed this: there should be one mutex for all unlocks, because we can't send several unlock txns from one chain simultaneously.

E.g., processor's instance for Ethereum can have batches of orders to be unlocked in Solana, Avalanche, etc... but it can send only one `batchUnlock` at a time because it uses only one account (`unlockAccount`).